sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -1279,7 +1279,7 @@ def _snapshot_name_version_filter(
         batches = self._snapshot_batches(name_versions)
 
         if not name_versions:
-            return exp.false()
+            yield exp.false()
```

Comment:
Mypy treated this as an "unexpected return value", because the return type of the method is an iterator (due to the `yield`s). Are there any weird semantics in place that I may have missed here? cc @tobymao.

Reply:
ah yea should be good i think